ADAL/src/cache/ADTokenCache.m

```diff
@@ -69,6 +69,45 @@ - (void)dealloc
     SAFE_ARC_SUPER_DEALLOC();
 }
 
+- (id)cacheObjectForKey:(id<NSCopying>)key
+{
+    id object = nil;
+    @synchronized(self)
+    {
+        object = _cache[key];
+    }
+    return object;
+}
+
+- (void)setCache:(NSMutableDictionary *)newCache
+{
+    @synchronized(self)
+    {
+        SAFE_ARC_RELEASE(_cache);
+        _cache = newCache;
+        SAFE_ARC_RETAIN(_cache);
+    }
+}
+
+- (NSMutableDictionary *)cacheCopy
+{
+    NSMutableDictionary *copy = nil;
+    @synchronized(self)
+    {
+        copy = [_cache mutableCopy];
+    }
+    SAFE_ARC_AUTORELEASE(copy);
```

Comment:
Yeah, "newObject", "copyObject", or "initObject" return owned objects. So, this method name should be safe. The broader point (that including the word "copy" in the method name is weird) still stands, though. I'm not really sure how to get around it unless we want to pollute this method with a bunch of `@synchronize`s.